src/taipy/core/data/abstract_file.py

```diff
@@ -26,3 +28,19 @@ def _build_path(self, storage_type):
         if not dir_path.exists():
             dir_path.mkdir(parents=True, exist_ok=True)
         return dir_path / f"{self.id}.{self.__EXTENSION_MAP.get(storage_type)}"
+
+    @staticmethod
+    def _check_and_update_preserve_file(old_path: Optional[str] = None, new_path: Optional[str] = None):
+        if preserve_file_path := os.getenv("TAIPY_PRESERVE_FILE_PATH"):
+            if old_path and pathlib.Path(preserve_file_path).is_file():
+                with open(preserve_file_path, "r+") as f:
+                    lines = f.readlines()
+                    old_path = f"{old_path}\n"
+                    if old_path in lines:
+                        f.seek(0)
+                        f.truncate()
+                        lines.remove(old_path)
+                        f.writelines(lines)
+            if new_path:
+                with open(preserve_file_path, "a") as f:
+                    f.write(f"{new_path}\n")
```

Comment:
That is a minor non blocking remark, but I would create 3 different methods to match the three use cases.
- append_preserve_file(path)
- remove_from_preserve_file(oldpath, newpath)
- replace_from_preserve_file(path)

I believe it is a way to self-document the code. you don't need to read the method to understand what it does.